parsl/monitoring/radios/udp_router.py

```diff
@@ -137,3 +148,53 @@ def udp_router_starter(*,
             router.start()
         except Exception:
             router.logger.exception("UDP router start exception")
+
+
+class UDPRadioReceiver():
+    def __init__(self, *, process: SpawnProcessType, exit_event: EventType, port: int) -> None:
+        self.process = process
+        self.exit_event = exit_event
+        self.port = port
+
+    def close(self) -> None:
+        self.exit_event.set()
+        join_terminate_close_proc(self.process)
+
+
+def start_udp_receiver(*,
+                       monitoring_messages: Queue,
+                       port: Optional[int],
+                       logdir: str,
+                       debug: bool) -> UDPRadioReceiver:
+
+    udp_comm_q: Queue[Union[int, str]]
+    udp_comm_q = SizedQueue(maxsize=10)
+
+    router_exit_event = SpawnEvent()
+
+    router_proc = SpawnProcess(target=udp_router_starter,
+                               kwargs={"comm_q": udp_comm_q,
+                                       "resource_msgs": monitoring_messages,
+                                       "exit_event": router_exit_event,
+                                       "udp_port": port,
+                                       "run_dir": logdir,
+                                       "logging_level": logging.DEBUG if debug else logging.INFO,
+                                       },
+                               name="Monitoring-UDP-Router-Process",
+                               daemon=True,
+                               )
+    router_proc.start()
+
+    try:
+        udp_comm_q_result = udp_comm_q.get(block=True, timeout=120)
+        udp_comm_q.close()
+        udp_comm_q.join_thread()
+    except queue.Empty:
```

Comment:
Should there be any provision for if `q_result` is not an `int`eger?  Or an integer in the valid range?  In a future type case, would reduce a debugging session to reading a log message.

Reply:
Should've been more direct in my question.  "That."